qiskit/extensions/standard/u1.py

```diff
@@ -15,43 +15,37 @@
 from qiskit.circuit import QuantumCircuit
 from qiskit.circuit import QuantumRegister
 from qiskit.circuit.decorators import _op_expand
-from qiskit.dagcircuit import DAGCircuit
 from qiskit.extensions.standard.u3 import U3Gate
 
 
 class U1Gate(Gate):
     """Diagonal single-qubit gate."""
 
-    def __init__(self, theta, qubit, circ=None):
+    def __init__(self, theta):
         """Create new diagonal single-qubit gate."""
-        super().__init__("u1", [theta], [qubit], circ)
+        super().__init__("u1", 1, [theta])
 
-    def _define_decompositions(self):
-        decomposition = DAGCircuit()
+    def _define(self):
+        definition = []
         q = QuantumRegister(1, "q")
-        decomposition.add_qreg(q)
         rule = [
-            U3Gate(0, 0, self.params[0], q[0])
+            (U3Gate(0, 0, self.params[0]), [q[0]], [])
         ]
         for inst in rule:
-            decomposition.apply_operation_back(inst)
-        self._decompositions = [decomposition]
+            definition.append(inst)
+        self.definition = definition
 
     def inverse(self):
         """Invert this gate."""
         self.params[0] = -self.params[0]
-        self._decompositions = None
+        self.definition = None
```

Comment:
`inverse` should return a new gate for the inverse, rather than modify the current gate.

Reply:
i agree. they all now return a new gate. I just removed the inplace option as it was not well-defined for standard gates.